scripts/ci/docker_upgrade.sh

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env bash
+set -e
+
+function error() {
+  echo >&2 "ERROR: ${1} Aborting." && false
+}
+
+function warning() {
+  echo >&2 "WARNING: ${1}"
+}
+
+[[ $# == 3 ]] || error "Expected exactly 3 parameters: ${0} <IMAGE_NAME> <IMAGE_VARIANT> <DOCKER_REPOSITORY>"
+
+IMAGE_NAME=${1}
+IMAGE_VARIANT=${2}
+DOCKER_REPOSITORY=${3}
+
+check_version() {
+  echo "-- check_version"
+
+  DOCKERFILE="scripts/docker/${IMAGE_NAME}/Dockerfile.${IMAGE_VARIANT}"
+
+  # continue running the whole workflow only if corresponding Dockerfile was changed.
+  if ! git diff --name-only origin/develop HEAD  -- | grep -e "^${DOCKERFILE}\$"
```

Comment:
`$DOCKERFILE` contains dots and could potentially contain other special regex characters in the future. Using it like that will result in false-positives.

Reply:
@aarlt
```
if git diff --quiet develop..HEAD "${DOCKERFILE}"
then
```

Reply:
good catch!